sycl/include/sycl/detail/pi.h

```diff
@@ -1163,6 +1172,12 @@ __SYCL_EXPORT pi_result piQueueCreate(pi_context context, pi_device device,
 __SYCL_EXPORT pi_result piextQueueCreate(pi_context context, pi_device device,
                                          pi_queue_properties *properties,
                                          pi_queue *queue);
+/// \param properties points to a zero-terminated array of extra data describing
+/// desired queue properties. Format is
+///  {[PROPERTY[, property-specific elements of data]*,]* 0}
+__SYCL_EXPORT pi_result piextQueueCreate2(pi_context context, pi_device device,
+                                          pi_queue_properties *properties,
+                                          pi_queue *queue);
```

Comment:
Could you please clarify how this API is different from `piextQueueCreate`?

Reply:
The decision was made to have different default behavior for the immediate command list setting for existing binaries and code recompiled with the 2023.2 and later releases. The version "2" APIs are designed to be able to tell whether the call into the runtime comes from existing binaries or recompiled binaries.

Reply:
But the call to `piextQueueCreate` is done from the SYCL RT, not from the user code. Do we consider situation when SYCL RT is old one while the plugin is new one?

Reply:
I haven't been reading the larger issue here, but I noticed this:

> Do we consider situation when SYCL RT is old one while the plugin is new one?

I'm curious about this.  Do we really have this situation?  I thought we packaged and shipped the SYCL RT together with the plugins, so they always match.

Reply:
The SYCL RT and L0 plugin have to be in sync, i.e. from the same release. (Even today, just setting LD_LIBRARY_PATH to a directory that contains libpi does not work. Both libsycl and libpi have to be found in the same place. That hints strongly at them having to be in sync).

Reply:
> The version "2" APIs are designed to be able to tell whether the call into the runtime comes from existing binaries or recompiled binaries.
> Both libsycl and libpi have to be found in the same place. That hints strongly at them having to be in sync).

So, then do not see how we can get calls from existing or recompiled binaries with the same release. We either have "old" release where SYCL RT always calls `piextQueueCreate` or a new release where SYCL RT always calls `piextQueueCreate2`. Am I missing something?
Anyway, it's minor, please feel free to ignore the comment.

Reply:
Have the same concern as @romanovvlad but I don't think it's minor. I don't understand how this `piextQueueCreate2` helps us to route existing binaries differently from re-compiled ones.